src/deploymentTemplate.ts

```diff
@@ -0,0 +1,149 @@
+                                items.push({ label: parsed[i].name, description: null });
+                            }
+                        }
+                        vscode.window.showQuickPick(items).then(selection => {
+                            // the user canceled the selection
+                            if (!selection) {
+                                return;
+                            }        
+                            
+                            __this.retrieveTemplate(selection.label);
+                        });
+                    });
+                } else {
+                    vscode.window.showErrorMessage("Error: " + response.statusCode + " " + response.statusMessage);
+                }
+
+            }).on('error', function(e) {
+                vscode.window.showErrorMessage("Error: " + e);
+            });
+        
+    }
+
+    public retrieveTemplate(templateName: string) {
+        var http = require('https');
+        let __this = this;
+        let repo: string = utilities.getCodeConfiguration<string>(null, Constants.Config_deploymentTemplatesGitHubRepo);
+
+            http.get({
+                host: Constants.GitHubRawContentHost,
+                path: '/' + repo + '/master/' + templateName + '/azuredeploy.json',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
+                }, function(response) {
+                    if (response.statusMessage == "OK") {
+                        var body = '';
+                        response.on('data', function(d) {
+                            body += d;
+                        });
+                        response.on('end', function() {
+                            try {
+                                var parsed = JSON.parse(body);
+
+                                __this.createPlaybookFromTemplate("https://" + Constants.GitHubRawContentHost + "/" + repo + "/master/" + templateName + "/azuredeploy.json",
+                                                                parsed);
+                            } catch (e) {
+                                vscode.window.showErrorMessage("Failed to parse 'azuredeploy.json'");
+                            }
+                        });
+                    } else if (response.statusCode == 404) {
+                        vscode.window.showErrorMessage("Template file 'azuredeploy.json' not found.");
+                    } else {
+                        vscode.window.showErrorMessage("Error: " + response.statusCode + " " + response.statusMessage);
+                    }
+                }).on('error', function(e) {
+                    vscode.window.showErrorMessage("Error: " + e);
+                });
+    }
+
+    public createPlaybookFromTemplate(location: string, template: object) {
+        let __this = this;
+
+        TelemetryClient.sendEvent('deploymenttemplateinserted', { 'template': location });
```

Comment:
let's keep event name consisnten: 'deployment template', but add 'inserted' as one of property: such as {'action':'templateinserted', 'template':'location'}